ogr/ogrsf_frmts/aivector/ograivectordriver.cpp

```diff
@@ -0,0 +1,147 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  Artificial Intelligence powered driver
+ * Author:   Even Rouault, <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2024, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "ogrsf_frmts.h"
+
+/************************************************************************/
+/*                       OGRAIVectorIdentify()                          */
+/************************************************************************/
+
+static int OGRAIVectorIdentify(GDALOpenInfo *poOpenInfo)
+{
+    return STARTS_WITH(poOpenInfo->pszFilename, "AIVector:") ||
+           poOpenInfo->IsSingleAllowedDriver("AIVector");
+}
+
+/************************************************************************/
+/*                         OGRAIVectorOpen()                            */
+/************************************************************************/
+
+static GDALDataset *OGRAIVectorOpen(GDALOpenInfo *poOpenInfo)
+{
+    if (!OGRAIVectorIdentify(poOpenInfo))
+        return nullptr;
+
+    class MyLayer final : public OGRLayer,
+                          public OGRGetNextFeatureThroughRaw<MyLayer>
+    {
+        OGRFeatureDefn *m_poLayerDefn = nullptr;
+        bool m_bReturnedFeature = false;
+
+        CPL_DISALLOW_COPY_ASSIGN(MyLayer)
+
+      public:
+        MyLayer()
+        {
+            m_poLayerDefn = new OGRFeatureDefn("result");
+            SetDescription(m_poLayerDefn->GetName());
+            m_poLayerDefn->Reference();
+            OGRFieldDefn oFieldDefn("name", OFTString);
+            m_poLayerDefn->AddFieldDefn(&oFieldDefn);
+            OGRSpatialReference *poSRS = new OGRSpatialReference(
+                "GEOGCS[\"I don't know\",\n"
```

Comment:
@jjimenezshaw Should we contact the OGC Naming Authority to register this one?

Reply:
"I don't care" 😜